serum_test.go

```diff
@@ -0,0 +1,114 @@
+package serum_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/serum-errors/go-serum"
+)
+
+func TestErrorsIs(t *testing.T) {
+	nonSerumErr := fmt.Errorf("non-serum")
+	nonSerumWrapped := fmt.Errorf("wrap: %w", nonSerumErr)
+	t.Run("sanity check", func(t *testing.T) {
+		if !errors.Is(nonSerumWrapped, nonSerumErr) {
+			t.Fatal("fmt.Errorf wrapping should be fine")
+		}
+	})
+	eqSynth := func(t *testing.T, a, b error, expectEqual bool) {
+		ai, ok := a.(serum.ErrorInterface)
+		if !ok {
+			t.Fatalf("%T is not a a serum.ErrorInterface", a)
+		}
+		bi, ok := b.(serum.ErrorInterface)
+		if !ok {
+			t.Fatalf("%T is not a a serum.ErrorInterface", b)
+		}
+		as := serum.SynthesizeString(ai)
+		bs := serum.SynthesizeString(bi)
+		if (as == bs) != expectEqual {
+			t.Fatalf("unexpected message synthesis:\n%s\n%s", as, bs)
+		}
+	}
+	t.Run("equivalent serum errors", func(t *testing.T) {
+		// should fail if !errors.Is
+		t.Run("with only codes", func(t *testing.T) {
+			a := serum.Error("test")
+			b := serum.Error("test")
+			if !errors.Is(a, b) {
+				t.Fatal("errors should be equivalent")
+			}
+			eqSynth(t, a, b, true)
+		})
+		t.Run("with literal messages", func(t *testing.T) {
+			a := serum.Error("test", serum.WithMessageLiteral("a thing!"))
+			b := serum.Error("test", serum.WithMessageLiteral("a thing!"))
+			if !errors.Is(a, b) {
+				t.Fatal("errors should be equivalent")
+			}
+			eqSynth(t, a, b, true)
+		})
+		t.Run("with untemplated details", func(t *testing.T) {
+			a := serum.Error("test", serum.WithMessageLiteral("a thing!"), serum.WithDetail("foo", "bar"))
+			b := serum.Error("test", serum.WithMessageLiteral("a thing!"), serum.WithDetail("bar", "grill"))
+			if !errors.Is(a, b) {
+				t.Fatal("we don't care about details which don't synthesize into the message")
+			}
+			eqSynth(t, a, b, true)
+		})
```

Comment:
This one is an interesting opinion and I'm not sure I'd expect it.

I looked briefly back to https://pkg.go.dev/errors#Is to see if there's any guidance there on whether the function is expected to be approximate or exact, and don't find any very clear treatment of it, so I'm not sure what to think will be minimally surprising to others...

Reply:
Today I'm leaning more towards _serialization_ equivalence instead of message equivalence. It might be weird either way. I'm going to change it and add some more comments. ¯\\_(ツ)_/¯

Reply:
Yeah, I think I concur with serialization equivalence being likely to be the least surprising.

(And if someone wants to only match on the code, then that is also fine... but that's what `serum.Code(err) ==` accomplishes already.  So the bases are both covered.)

Reply:
I think there's a good argument for an `IsCode` function which will return true for anything wrapped, but that's not hard to implement independently.